### PR TITLE
scripted prototype for seqr interactions

### PR DESCRIPTION
**Pull request by MattWellie**
# Fixes

  - relevant to #152 

This is not polished or robust, it's just a proof of concept

## Proposed Changes

  - Dependent on https://github.com/populationgenomics/seqr/pull/167
  - prototype post-aip-run script so that (if the `@service_account_access` auth was live):
  
1. query metamist for the correct seqr project to use, or die if not found
2. query seqr for CPGid -> seqr family ID   lookup
3. query seqr for all current flagged variants in the project
4. digest a set of AIP results
5. find all results in the AIP set which don't already exist as flags
6. push each of those flags as a new saved variant

I've played around with this on a local seqr deployment, and tested most of these methods/api calls against that instance. Without an elastic connection it's not been possible to confirm that the pushed tag work perfectly

## Not done: 

There's some many2many mappings behind the scenes, so the same tag can be shared by 2 families , or a variant can be linked to multiple tags. This is a basic implementation, but there's some slightly more involved syntax that could be used to update/extend existing tags instead of always creating new ones.

Also a fair bit of this logic could be done on the seqr ORM side, but I gave up on that because tracing joins on the backend was giving me a migraine.

## Checklist

- [ ] Related Issue created
- [ ] Tests covering new change
- [ ] Linting checks pass

### CODE REVIEW

File: reanalysis/seqr_interactions.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+    # WALRUS
+    if results := len(projects) != 1:
+        raise Exception(f'Expected one project for {project_dataset}, found {results}')
+
+    project = projects[0]
+
+    if proj_seqr_key in project['meta'] and project['meta'].get('is_seqr'):
+        return project['meta'].get(proj_seqr_key)
+
+    return None
```
Comment (illusional) [inline reanalysis/seqr_interactions.py:91]: You're already getting seqr projects above:
```suggestion
    return project['meta'].get(proj_seqr_key)
```

File: reanalysis/seqr_interactions.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+        # now check to remove redundancy
+        existing = extant_flags[family_id]
+
+        for variant_id, tags in variants.items():
+            for tag in tags - existing.get(variant_id, set()):
+                flags_to_add.append(
+                    {'family': family_id, 'tag': tag, 'variant': variant_id}
+                )
+            continue
```
Comment (illusional) [inline reanalysis/seqr_interactions.py:288]: Redundant continue?

**Review by illusional — COMMENTED** [review]
Looking good!

File: reanalysis/seqr_interactions.py (hunk 1 above)
Comment (MattWellie) [reply reanalysis/seqr_interactions.py:91]: Heh, I was wondering if there was any possibility of having a populated project and `is_seqr=False`, but that wouldn't make any sense...

File: reanalysis/seqr_interactions.py (hunk 2 above)
Comment (MattWellie) [reply reanalysis/seqr_interactions.py:288]: Yup!

File: reanalysis/seqr_interactions.py [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+    proj_seqr_key = f'seqr-project-{get_config()["workflow"]["sequencing_type"]}'
+
+    # filter the returned projects
+    projects = [
+        proj
+        for proj in ProjectApi().get_seqr_projects()
+        if proj.get('name') == project_dataset
+    ]
+
+    # WALRUS
```
Comment (cassimons) [inline reanalysis/seqr_interactions.py:82]: IMHH the walrus should die in a fire 🔥 🦭 🔥

**Review by cassimons — APPROVED** [review]
Very cool!

**Comment by cassimons** [comment]
Can you retract a 'approve? Too much muscle memory...

**Comment by MattWellie** [comment]
Closing for now

**Pull request closed without merging**